File: lib/magician/numeric.rb
# Magician's extensions to the Numeric class (affects Integers and Floats).
class Numeric

  # Returns true if the number is evenly divisible by n. If n is equal to 0, it
  # returns false, since numbers cannot be divided by 0 in real number
  # arithmetic.
  #
  # @param [Numeric] n the number that this number (self) should be divided by
  #
  # @return [Boolean] true if the number is evenly divisible by n
  def divisible? n
    n.zero? ? false : modulo(n).zero?
  end

  # Converts the number from degrees to radians and returns the result.
  #
  # @return [Numeric] the number in radians
  def to_radians
    self * Math::PI / 180
  end

  # Converts the number from radians to degrees and returns the result.
  #
  # @return [Numeric] the number in degrees
  def to_degrees
    self * 180 / Math::PI
  end

end

File: src/magician/numeric.clj
(ns magician.numeric)

;;;;  Magician's extensions to the Numeric class (affects Integers and Floats).

(defn divisible?
  "Returns true if for numbers x and y, x is evenly divisible by y.

  If y is equal to 0, it returns false, since numbers cannot be divided by 0 in
  real number arithmetic."
  [x y]
  (if (= y 0)
    false
    (zero? (mod x y))))

(defn degrees->radians
  "Converts a number from degrees to radians."
  [n]
  (* n (/ Math/PI 180)))

(defn radians->degrees
  "Converts a number from radians to degrees."
  [n]
  (* n (/ 180 Math/PI)))
